frontend/packages/console-app/src/components/guided-tours/GuidedTourItem.tsx

```diff
@@ -1,39 +1,74 @@
 import * as React from 'react';
+import { Dispatch } from 'redux';
+import { connect } from 'react-redux';
 import { CatalogTile } from '@patternfly/react-catalog-view-extension';
-import { GuidedTourItem, GuidedTourCatalogItem } from './utils/guided-tour-typings';
+import { GuidedTourCatalogItem } from './utils/guided-tour-typings';
 import TourItemHeader from './TourItemHeader';
 import TourItemDescription from './TourItemDescription';
 import TourItemFooter from './TourItemFooter';
 import './GuidedTourItem.scss';
+import { toggleGuidedTourSidebarExpanded } from '../../redux/actions/guided-tour-actions';
+import { getTourStatus, setTourStatus } from './utils/guided-tour-utils';
+import { GuidedTourStatus } from './utils/guided-tour-status';
 
-type GuidedTourItemProps = GuidedTourCatalogItem;
+type DispatchProps = {
+  onClick: (tourId: string) => void;
+};
 
-const GuidedTourItem: React.FC<GuidedTourItemProps> = ({
+type GuidedTourItemProps = GuidedTourCatalogItem & DispatchProps;
+
+export const GuidedTourItem: React.FC<GuidedTourItemProps> = ({
   iconURL,
   altIcon,
+  id,
   name,
   description,
   status,
   active,
   duration,
   prerequisites,
   unmetPrerequisite,
-}) => (
-  <CatalogTile
-    iconImg={iconURL}
-    iconAlt={altIcon}
-    className="oc-guided-tour-item"
-    featured={active}
-    title={<TourItemHeader name={name} status={status} duration={duration} />}
-    description={
-      <TourItemDescription
-        description={description}
-        prerequisites={prerequisites}
-        unmetPrerequisite={unmetPrerequisite}
-      />
-    }
-    footer={<TourItemFooter unmetPrerequisite={unmetPrerequisite} status={status} />}
-  />
-);
+  onClick,
+}) => {
+  const handleClick = () => {
+    getTourStatus(id) !== GuidedTourStatus.COMPLETE &&
+      setTourStatus(id, GuidedTourStatus.IN_PROGRESS);
+    onClick(id);
+  };
+
+  return (
+    <CatalogTile
+      key={name}
+      iconImg={iconURL}
+      iconAlt={altIcon}
+      className="oc-guided-tour-item"
+      featured={active}
+      title={
+        <TourItemHeader
+          key={name}
+          name={name}
+          status={getTourStatus(id) || status}
+          duration={duration}
+        />
+      }
+      onClick={handleClick}
+      description={
+        <TourItemDescription
+          key={name}
+          description={description}
+          prerequisites={prerequisites}
+          unmetPrerequisite={unmetPrerequisite}
+        />
+      }
+      footer={<TourItemFooter key={name} unmetPrerequisite={unmetPrerequisite} status={status} />}
+    />
+  );
+};
+
+const mapPropsToState = (dispatch: Dispatch): DispatchProps => ({
+  onClick: (tourId: string) => dispatch(toggleGuidedTourSidebarExpanded(tourId)),
+});
+
+const ConnectedGuidedTourItem = connect<{}, DispatchProps>(null, mapPropsToState)(GuidedTourItem);
```

Comment:
`onClick` should be handled by the `GuidedTourCatalog` and passed in as a prop.

Reply:
done